src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotPassTypesByReference.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotPassTypesByReference : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1045";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceMessage), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceDescription), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Design,
```

Comment:
@mavasani Can we place this rule in a new category `LegacyDesign` to indicate that it is no longer a recommended design rule? I worry the rate of false positives would exceed the benefit provided by the rule.

Reply:
I have added this new category and applied it for this rule. Feel free to revert the commit if you don't want it.

Reply:
I discussed this with @mavasani, and concluded that the category-based configuration for rules will not apply automatically to rules that are disabled by default, which makes it less likely that this rule will be accidentally enabled. This behavior was already covered by the following test case, but @mavasani is updating the documentation for this test case to call out its importance for analyzers like this so we don't unintentionally change that behavior in the future.

https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs#L11603-L11608